lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -183,7 +185,9 @@ sub list_scheduled_ajax {
     );
 
     my @scheduled;
-    while (my $job = $scheduled->next) {
+    my $i = 0;
+    while (my $job = $scheduled->next and $i < $limit) {
+        $i++;
```

Comment:
Same as the above comment. Also, `complex_query` calls `_prepare_complex_query_search_args`, which already takes a `limit` parameter.